test/lib/storage_proposal_test.rb

```diff
@@ -103,21 +104,18 @@
       allow(proposal).to receive(:planned_devices).and_return(planned_devices)
     end
 
-    context "when the proposal was successful" do
+    context "when the proposal did not fail" do
       it "returns true" do
         expect(storage_proposal).to be_valid
       end
 
-      context "when no root partition is found in the proposal" do
-        let(:partition) { Y2Storage::Planned::Partition.new("/home", nil) }
-
-        it "returns false" do
-          expect(storage_proposal).to_not be_valid
+      context "but an issue was detected" do
+        before do
+          issues_list.add(:missing_root)
         end
 
-        it "registers the error" do
-          expect { storage_proposal.valid? }.to change { storage_proposal.issues? }
-            .from(false).to(true)
+        it "returns false" do
+          expect(storage_proposal).to be_valid
```

Comment:
I guess this should be `expect(storage_proposal).to_not be_valid`, so does the test pass?

Reply:
Oh, that's true. The test is wrong (so it pass, but it shouldn't). Fixed.